src/app/clients/add/formSections/NumberAdultsCard.tsx

```diff
@@ -0,0 +1,86 @@
+import React from "react";
+import FreeFormTextInput from "@/components/DataInput/FreeFormTextInput";
+import {
+    CardProps,
+    Error,
+    errorExists,
+    ErrorSetter,
+    errorText,
+    FieldSetter,
+    numberRegex,
+    Person,
+    PersonType,
+} from "@/components/Form/formFunctions";
+import GenericFormCard from "@/components/Form/GenericFormCard";
+import { selectChangeEventHandler } from "@/components/DataInput/inputHandlerFactories";
+
+const getNumberAdults = (
+    fieldSetter: FieldSetter,
+    errorSetter: ErrorSetter,
+    adults: Person[],
+    personType: PersonType
+): selectChangeEventHandler => {
+    return (event) => {
+        const input = event.target.value;
+        const newValue = adults;
+        const personIndex = newValue.findIndex((object) => object.personType === personType);
+
+        if (input === "") {
+            newValue[personIndex].quantity = 0;
+        } else if (input.match(numberRegex)) {
+            newValue[personIndex].quantity = parseInt(input);
+        } else {
+            newValue[personIndex].quantity = -1;
+        }
+
+        fieldSetter("adults", newValue);
+
+        const invalidAdultEntry = newValue.filter((value) => value.quantity === -1);
+        const nonZeroAdultEntry = newValue.filter((value) => value.quantity! > 0);
```

Comment:
`quantity` is an optional property in the `Person` interface since `children` have `quantity = undefined` (this is then set to `1` when the form is submitted). However, since we're dealing with `adults` here, their quantity will always be a number and so we can assert that it is never undefined in this case.